### PR TITLE
B2b smoke test reclaimation

### PR DESCRIPTION
**Pull request by durgesh-verma-spryker**
## PR Description
https://spryker.atlassian.net/browse/CC-21651

Reclamation
-  Create a Reclamation for some products of the order and process their statuses in the Reclamation details. Close reclamation

### CODE REVIEW

File: resources/steps/reclaimation_steps.robot [hunk 1]
```diff
@@ -0,0 +1,16 @@
+*** Settings ***
+Resource    ../common/common_zed.robot
+Resource    ../common/common.robot
+
+*** Variables ***
```
Comment (AntonShopin) [inline resources/steps/reclaimation_steps.robot:5]: don't mix variables with keywords. 
For variables create a separate file to describe this page (locators). Page Object pattern. 
Actions (keywords) should stay in '_steps' file

File: resources/steps/reclaimation_steps.robot [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+${all_product_checkbox}    id=check-all-orders
+${create_reclamation_button_locator}    xpath=//input[@type="submit"]
+*** Keywords ***
+zed: create reclaimation
```
Comment (AntonShopin) [inline resources/steps/reclaimation_steps.robot:9]: 'zed' should start from the upper case and you have a typo. "Zed: create reclamation"

This keyword is not 'smart' enough:
* with this keyword it should be possible to create a reclamation for all items or for some specific ones. Please refactor

File: resources/steps/reclaimation_steps.robot [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+    Click    ${all_product_checkbox}
+    Click    ${create_reclamation_button_locator}
+    Zed: flash message should be shown:    success
+
+Zed: select latest Reclamation
```
Comment (AntonShopin) [inline resources/steps/reclaimation_steps.robot:14]: well, I don't agree with this one, you just click on the first column to sort, but what if the table is already sorted by the first column? You need to check if it 'ascending' and only then click to sort to have 'descending'. If it's already 'descending' - do not click

File: tests/smoke/smoke_b2b.robot [hunk 4]
```diff
@@ -866,4 +867,29 @@ Refunds
     Zed: trigger all matching states inside this order:    Refund
     Zed: grand total for the order equals:    ${lastPlacedOrder}    €0.00
     [Teardown]    Run keywords    Zed: login on Zed with provided credentials:    ${zed_admin_email}
-    ...    AND    Zed: activate following discounts from Overview page:    20% off storage    10% off minimum order
\ No newline at end of file
+    ...    AND    Zed: activate following discounts from Overview page:    20% off storage    10% off minimum order
+
+Reclaimation
```
Comment (AntonShopin) [inline tests/smoke/smoke_b2b.robot:872]: Your tests should cover the following: 
* create reclamation for part of the order items, not the whole order
* view the details of the Reclamation and check that only selected items are inside this Reclamation 
* Close the reclamation and check that the state of the Reclamation is now 'closed'

**Review by AntonShopin — CHANGES_REQUESTED** [review]

File: resources/steps/reclaimation_steps.robot (hunk 1 above)
Comment (durgesh-verma-spryker) [reply resources/steps/reclaimation_steps.robot:5]: Done

File: resources/steps/reclaimation_steps.robot (hunk 2 above)
Comment (durgesh-verma-spryker) [reply resources/steps/reclaimation_steps.robot:9]: Fixed

File: resources/steps/reclaimation_steps.robot (hunk 3 above)
Comment (durgesh-verma-spryker) [reply resources/steps/reclaimation_steps.robot:14]: Fixed

File: resources/steps/reclaimation_steps.robot (hunk 1 above)
Comment (durgesh-verma-spryker) [reply resources/steps/reclaimation_steps.robot]: Fixed

File: resources/steps/reclaimation_steps.robot (hunk 2 above)
Comment (durgesh-verma-spryker) [reply resources/steps/reclaimation_steps.robot]: Fixed

File: resources/steps/reclaimation_steps.robot (hunk 3 above)
Comment (durgesh-verma-spryker) [reply resources/steps/reclaimation_steps.robot]: Fixed

File: tests/smoke/smoke_b2b.robot (hunk 4 above)
Comment (durgesh-verma-spryker) [reply tests/smoke/smoke_b2b.robot:872]: Fixed

File: resources/steps/reclaimation_steps.robot [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+ 
+Zed: reclamation table sorting in descending order
+   ${first}    Get Text    xpath=(//tbody/tr/td[@class="column-spy_sales_reclamation.id_sales_reclamation sorting_1"])[1]
+   ${second}    Get Text    xpath=(//tbody/tr/td[@class="column-spy_sales_reclamation.id_sales_reclamation sorting_1"])[2]
+   IF   '${first}'<'${second}'
+       Click    xpath=//th[@id="spy_sales_reclamation.id_sales_reclamation" and contains(text(),'#')]
+   END
+ 
+Zed: create reclaimation for part of order:
+   [Arguments]    ${sku}    ${reclaim}=1
```
Comment (AntonShopin) [inline resources/steps/reclaimation_steps.robot:22]: 1. This keyword should accept multiple SKUs, not just one
2. What is the role of `${reclaim}=1`, sorry, I didn't get the usage

File: resources/steps/reclaimation_steps.robot [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+   IF    '${env}' in ['b2b','b2c']
+       ${elementSelector}=    Set Variable    xpath=//table[@data-qa='order-item-list'][${reclaim}]/tbody//td//div[@class='sku'][contains(text(),'${sku}')]/ancestor::tr/td/input[@type="checkbox"]
+   ELSE
+      ${elementSelector}=    Set Variable    xpath=//table[@data-qa='order-item-list'][${reclaim}]/tbody//td//div[@class='sku'][contains(text(),'${sku}')]/ancestor::tr/td/input[@type="checkbox"]
```
Comment (AntonShopin) [inline resources/steps/reclaimation_steps.robot:26]: Both variables are exactly the same. Why did you put `IF` here?

File: resources/steps/reclaimation_steps.robot [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+   END
+   Check Checkbox    ${elementSelector}
+
+Zed: check details of latest created reclaimation:
```
Comment (AntonShopin) [inline resources/steps/reclaimation_steps.robot:30]: 1. please rename it. Looks like this keyword supposes to check that the latest reclamation contains some SKUs. From the current name, it's not clear
2. please refactor so it accepts multiple SKUs

File: resources/steps/reclaimation_steps.robot [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+    Wait Until Element Is Visible    ${first_view_reclaimation} 
+    Click    ${first_view_reclaimation} 
+    Page Should Contain Element    //div[@class="sku" and contains(text(),'${sku}')]
+
+Zed: close latest created reclaimation
+    Zed: reclamation table sorting in descending order
+    Wait Until Element Is Visible    ${close_button_reclaimation}
+    Click    ${close_button_reclaimation}
+
+Zed: check state of latest reclaimation:
```
Comment (AntonShopin) [inline resources/steps/reclaimation_steps.robot:42]: 1. Please rename it so it's clear what argument I should pass
2. This keyword doesn't work as your locator for the state is hardcoded for 'Closed' state. If the Reclamation is 'Open' - you locator shows 0 results

File: resources/pages/zed/zed_reclaimation_page.robot [hunk 9]
```diff
@@ -0,0 +1,6 @@
+*** Variables ***
+${all_product_checkbox}    id=check-all-orders
+${create_reclamation_button_locator}    xpath=//input[@type="submit"]
+${first_view_reclaimation}    (//a[contains(@href,"/sales-reclamation-gui/detail?id-reclamation")])[1]
+${close_button_reclaimation}    (//button[@type="submit"])[1]
+${state_reclaimation_locator}    (//span[contains(@class,"label-danger")])[1]
```
[the remark this answers is not in the document]
Comment (durgesh-verma-spryker) [reply resources/pages/zed/zed_reclaimation_page.robot:6]: Fixed

File: resources/steps/reclaimation_steps.robot [hunk 10]
```diff
@@ -0,0 +1,46 @@
+*** Settings ***
+Resource   ../common/common_zed.robot
+Resource   ../common/common.robot
+Resource   ../../resources/pages/zed/zed_reclaimation_page.robot
+
+*** Keywords ***
+Zed: check checkbox of reclaimation of whole order
```
[the remark this answers is not in the document]
Comment (durgesh-verma-spryker) [reply resources/steps/reclaimation_steps.robot:7]: Fixed

File: resources/steps/reclaimation_steps.robot [hunk 11]
```diff
@@ -0,0 +1,46 @@
...
+   Click    ${all_product_checkbox}
+ 
+Zed: submit reclaimation
+   Click    ${create_reclamation_button_locator}
+   Zed: flash message should be shown:    success
+ 
+Zed: reclamation table sorting in descending order
+   ${first}    Get Text    xpath=(//tbody/tr/td[@class="column-spy_sales_reclamation.id_sales_reclamation sorting_1"])[1]
+   ${second}    Get Text    xpath=(//tbody/tr/td[@class="column-spy_sales_reclamation.id_sales_reclamation sorting_1"])[2]
```
[the remark this answers is not in the document]
Comment (durgesh-verma-spryker) [reply resources/steps/reclaimation_steps.robot:16]: Fixed, as per your suggestion

File: resources/steps/reclaimation_steps.robot [hunk 12]
```diff
@@ -0,0 +1,46 @@
...
+
+*** Keywords ***
+Zed: check checkbox of reclaimation of whole order
+   Click    ${all_product_checkbox}
+ 
+Zed: submit reclaimation
+   Click    ${create_reclamation_button_locator}
+   Zed: flash message should be shown:    success
+ 
+Zed: reclamation table sorting in descending order
```
[the remark this answers is not in the document]
Comment (durgesh-verma-spryker) [reply resources/steps/reclaimation_steps.robot:14]: Fixed, as suggested

File: resources/steps/reclaimation_steps.robot (hunk 5 above)
Comment (durgesh-verma-spryker) [reply resources/steps/reclaimation_steps.robot:22]: Fixed

File: resources/steps/reclaimation_steps.robot (hunk 6 above)
Comment (durgesh-verma-spryker) [reply resources/steps/reclaimation_steps.robot:26]: Fixed

File: resources/steps/reclaimation_steps.robot (hunk 7 above)
Comment (durgesh-verma-spryker) [reply resources/steps/reclaimation_steps.robot:30]: Fixed

File: resources/steps/reclaimation_steps.robot (hunk 8 above)
Comment (durgesh-verma-spryker) [reply resources/steps/reclaimation_steps.robot:42]: Fixed

**Pull request closed without merging**